### PR TITLE
Exclude samples for interactive problems.

### PR DESCRIPTION
**Pull request by eldering**
(cherry picked from commit 0cc0da082f9d96a0cc653755d4e22fb837a3c84d)

### CODE REVIEW

**Comment by Kevinjil** [comment]
Wouldn't it be nicer if we do not allow downloading a sample zip for interactive problems, instead of supplying an empty zip?

**Comment by meisterT** [comment]
This code path is the overall zip with all samples, and we still include the attachments, so it is not empty.

**Review by meisterT — APPROVED** [review]

File: webapp/src/Service/DOMJudgeService.php [hunk 1]
```diff
@@ -881,7 +881,10 @@ public function getSamplesZipForContest(Contest $contest): StreamedResponse
 
         /** @var ContestProblem $problem */
         foreach ($contest->getProblems() as $problem) {
-            $this->addSamplesToZip($zip, $problem, $problem->getShortname());
+            // We don't include the samples for interactive problems.
+            if (!$problem->getProblem()->getCombinedRunCompare()) {
```
[the remark this answers is not in the document]
Comment (meisterT) [reply webapp/src/Service/DOMJudgeService.php:885]: Who says you are in the minority :-)

I think it is a good idea to add a function alias.

[the remark this answers is not in the document]
Comment (eldering) [reply webapp/src/Service/DOMJudgeService.php:885]: I mentioned IRL that I don't like adding that alias, as it hides that under the hood this `CombinedRunCompare`, which I _think_ is not necessarily exactly the same as an interactive compare script. I think it's better to leave that explicit than make an alias function look like a clean expression of the problem being an inteactive problem. OTOH, if this really 1-1 means that this is an interactive compare script and thus problem, then I think we should just rename that table column and this associated getter.

[the remark this answers is not in the document]
Comment (meisterT) [reply webapp/src/Service/DOMJudgeService.php:885]: Let's address/discuss this outside of this PR.

**Pull request merged**